daemon/k8s_watcher.go

```diff
@@ -334,31 +337,46 @@ func (d *Daemon) addK8sNetworkPolicy(obj interface{}) {
 }
 
 func (d *Daemon) updateK8sNetworkPolicy(oldObj interface{}, newObj interface{}) {
-	scopedLog := log.WithFields(log.Fields{
-		"obj.old": logfields.Repr(oldObj),
-		"obj.new": logfields.Repr(newObj),
-	})
-
-	k8sNP, ok := newObj.(*networkingv1.NetworkPolicy)
-	if ok {
-		scopedLog = scopedLog.WithField(logfields.K8sAPIVersion, k8sNP.APIVersion)
+	// We don't need to deepcopy the object since we are transforming it to
+	// a Cilium Network Policy rule.
+	oldk8sNP, ok := oldObj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".old", logfields.Repr(oldObj)).
+			Warn("Ignoring invalid k8s NetworkPolicy modification")
+		return
+	}
+	newk8sNP, ok := newObj.(*networkingv1.NetworkPolicy)
+	if !ok {
+		log.WithField(logfields.Object+".new", logfields.Repr(newk8sNP)).
+			Warn("Ignoring invalid k8s NetworkPolicy modification")
+		return
 	}
+	log.WithFields(log.Fields{
+		logfields.K8sAPIVersion:                 oldk8sNP.APIVersion,
+		logfields.K8sNetworkPolicyName + ".old": oldk8sNP.Name,
+		logfields.K8sNamespace + ".old":         oldk8sNP.Namespace,
+		logfields.K8sNetworkPolicyName + ".new": newk8sNP.Name,
+		logfields.K8sNamespace + ".new":         newk8sNP.Namespace,
+	}).Debug("Modified policy")
```

Comment:
You haven't actually modified the policy by the time this debug log is being written

Reply:
Since we can't know if a policy was actually modified or not I've change it to `Received policy update`